db/document.go

```diff
@@ -306,6 +305,15 @@ func (doc *document) UnmarshalWithXattr(data []byte, xdata []byte) error {
 		return err
 	}
 	// Unmarshal document body
+	if len(data) > 0 {
```

Comment:
I think the flow might be slightly clearer if there's an early return and dropping the else block

```
	// Unmarshal document body
	if len(data) > 0 {
		return doc.unmarshalBody(data)
	} 
	
        // If there's an xattr but no body, set body as {"_deleted":true} to align with non-xattr handling
	doc.body = Body{}
	doc.body["_deleted"] = true
	return nil
	```